js/packages/teams-ai/src/models/OpenAIModel.ts

```diff
@@ -399,20 +485,45 @@ export class OpenAIModel implements PromptCompletionModel {
     /**
      * @private
      * @param {string} model - Model to use.
-     * @param {Message<string>[]} messages - Messages to send.
+     * @param {ChatCompletionMessageParam[]} messages - Messages to send.
      * @param {PromptTemplate} template Prompt template being used.
      * @returns {ChatCompletionCreateParams} Chat completion parameters.
      */
     private getChatCompletionParams(
         model: string,
-        messages: Message<string>[],
+        messages: ChatCompletionMessageParam[],
         template: PromptTemplate
     ): ChatCompletionCreateParams {
+        // Validate Tools Augmentation
+        const isToolsAugmentation =
+            template.config.augmentation && template.config.augmentation?.augmentation_type == 'tools';
+
+        const chatCompletionTools = isToolsAugmentation
+            ? template.actions?.map((action) => {
+                  return {
+                      type: 'function',
+                      function: {
+                          name: action.name,
+                          description: action.description ?? '',
+                          parameters: action.parameters ?? {}
+                      }
+                  } as ChatCompletionTool;
+              })
+            : [];
+
+        const completion = {
+            ...template.config.completion,
+            tool_choice: isToolsAugmentation ? (template.config.completion.tool_choice ?? 'auto') : undefined,
+            tools: chatCompletionTools,
+            // Only include parallel_tool_calls if tools are enabled and the template has it set; otherwise, it will default to true without being added to the API call
+            parallel_tool_calls: isToolsAugmentation ? template.config.completion.parallel_tool_calls : undefined
```

Comment:
does this need to be

`template.config.completion.parallel_tool_calls ?? True`

Reply:
Nope, the comment explains it